themes/default/content/registry/packages/aws/installation-configuration.md

```diff
@@ -208,3 +208,73 @@ Use `pulumi config set aws:<option>` or pass options to the [constructor of `new
 | `skipRegionValidation` | Optional | Skip static validation of region name. Used by users of alternative AWS-like APIs or users w/ access to regions that are not public (yet). |
 | `skipRequestingAccountId` | Optional | Skip requesting the account ID. Used for AWS API implementations that do not have IAM/STS API and/or metadata API. |
 | `token` | Optional | Use this to set an MFA token. It can also be sourced from the `AWS_SESSION_TOKEN` environment variable. |
+
+## Centralize your configuration
+
+While you can create your AWS configurations locally, you also have the option to centralize your configurations using [Pulumi ESC (Environments, Secrets, and Configuration)](https://www.pulumi.com/docs/pulumi-cloud/esc/). Using this service will enable you to do things like run AWS CLI commands or run the `pulumi up` command with dynamically generated credentials, removing the need to configure and manage your credentials locally.
+
+To do this, you will need to complete the following steps:
+
+### Configure OIDC between Pulumi and AWS
+
+Refer to the [Configuring OpenID Connect for AWS Guide](https://www.pulumi.com/docs/pulumi-cloud/oidc/aws/) for the step-by-step process on how to do this. Note that when adding your configuration to your environment file, you can also define your AWS region in the `environmentVariables` section:
+
+```yaml
+values:
+  aws:
+    login:
+      fn::open::aws-login:
+        oidc:
+          duration: 1h
+          roleArn: <your-oidc-iam-role-arn>
+          sessionName: pulumi-environments-session
+  environmentVariables:
+    AWS_ACCESS_KEY_ID: ${aws.login.accessKeyId}
+    AWS_SECRET_ACCESS_KEY: ${aws.login.secretAccessKey}
+    AWS_SESSION_TOKEN: ${aws.login.sessionToken}
+    AWS_REGION: <YOUR_AWS_REGION> # e.g.`ap-south-1`
+```
+
+### [Optional] Move Pulumi config to your ESC environment
```

Comment:
Is a separate section needed here? I ask because this section is really only conveying that you can add a `pulumiConfig` block if you want -- which, I now that I think of it, could also be used to apply _all_ of these settings (ID, key, token, and region), rather than just region. Right? This, for example, is another way of saying the same thing:

```yaml
# myorg/myenv
aws:
    login:
      fn::open::aws-login:
        oidc:
          duration: 1h
          roleArn: <your-oidc-iam-role-arn>
          sessionName: pulumi-environments-session

values:
  pulumiConfig:
    aws:accessKey: ${aws.login.accessKeyId}
    aws:secretKey: ${aws.login.secretAccessKey}
    aws:token: ${aws.login.sessionToken}
    aws:region: ${aws.region}
```

What do you think about showing both of these in the same section, along with the content of `Import your environment`, since it all goes together (and again is collectively just one more way to configure the AWS provider)?

Reply:
I'm glad you pointed this out because I was actually thinking of updating this section so that the example I show in the `pulumiConfig` part is not specific to the credentials variables. I want to let folks know that "hey you can include other Pulumi config in here, not just AWS credentials and not just things related to AWS".

I would be happy with something like:

```yaml
values:
  aws:
    login:
      fn::open::aws-login:
        oidc:
          duration: 1h
          roleArn: <your-oidc-iam-role-arn>
          sessionName: pulumi-environments-session
  environmentVariables:
    AWS_ACCESS_KEY_ID: ${aws.login.accessKeyId}
    AWS_SECRET_ACCESS_KEY: ${aws.login.secretAccessKey}
    AWS_SESSION_TOKEN: ${aws.login.sessionToken}
    AWS_REGION: <YOUR_AWS_REGION>
  pulumiConfig: # exposes Pulumi config values to the Pulumi CLI
    project:environment: 'dev'
    aws:roleArn: 'arn:aws:iam::058111598222:role/OrganizationAccountAccessRole'
    aws:dynamodbEndpoint: 'dynamodb.us-east-2.amazonaws.com'
```

[note that adds your comment about not having to expose environment variables if it is not needed for the workflow] For example:


```yaml
values:
  aws:
    login:
      fn::open::aws-login:
        oidc:
          duration: 1h
          roleArn: <your-oidc-iam-role-arn>
          sessionName: pulumi-environments-session
  pulumiConfig: 
    aws:region: <YOUR_AWS_REGION>
    aws:accessKey: ${aws.login.accessKeyId}
    aws:secretKey: ${aws.login.secretAccessKey}
    aws:token: ${aws.login.sessionToken}
    project:environment: 'dev'
    aws:roleArn: 'arn:aws:iam::058111598222:role/OrganizationAccountAccessRole'
    aws:dynamodbEndpoint: 'dynamodb.us-east-2.amazonaws.com'
```